lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderServiceImpl.java

```diff
@@ -28,190 +31,320 @@
 
   private Context context;
 
-  private boolean mockMode;
-  private LocationEngine locationEngine;
-  private Map<LocationListener, LocationRequest> listenerToRequest;
-  private Map<PendingIntent, LocationRequest> intentToRequest;
-  private Map<LocationCallback, LocationRequest> callbackToRequest;
-  private Map<LocationCallback, Looper> callbackToLooper;
+  private Map<LostApiClient, Map<LocationListener, LocationRequest>> listenerToRequest;
```

Comment:
We may need to rename this class to `CartographyServiceImpl` because of all these maps 😉 

But seriously I wonder if there is a way we can simplify this design. What do you think about expanding the scope of the `ClientManager` to track which `LocationRequests` are associate with which clients?